tests/ltp/batch.mk

```diff
@@ -1,4 +1,10 @@
-include ../../common.mk
+ifneq (,$(wildcard ../../common.mk)) 
```

Comment:
Can you merge all those ifs into a single one? I don't think there's a need to check the same thing multiple times. Also, if you just have a single condition and then the else branch (not else if) it would be enough as well.

Reply:
I don't understand what you mean. Do you mean you have something like

```
if A && B && C
x
else if D && E && F
y
else
error couldnt find one of these 3 file
end
```

Reply:
No, just `ifneq (,$(wildcard ../../common.mk)) `. That's a proxy of knowing where you are in the hierarchy.

Reply:
Sorry, I'm not understanding the change you are suggesting.

Reply:
```
ifneq (,$(wildcard ../../common.mk)) 
  include ../../common.mk
  LTP_TEST_SCRIPT=../run_ltp_test.sh
  BUILDENV_SCRIPT=../buildenv.sh
else
  include ../common.mk
  LTP_TEST_SCRIPT=./run_ltp_test.sh
  BUILDENV_SCRIPT=./buildenv.sh
endif

ALPINE_MAJOR=3.8
ALPINE_VERSION=3.8.0
ALPINE_ARCH=x86_64

ROOT_FS=sgxlkl-miniroot-fs.img.master
ROOT_FS_FRESH_COPY=sgxlkl-miniroot-fs.img
ALPINE_TAR=alpine-minirootfs.tar.gz
MOUNTPOINT=/media/ext4disk
IMAGE_SIZE_MB=1500

# 1 hours timeout for ltp test execution
LTP_TEST_EXEC_TIMEOUT=3600

ESCALATE_CMD=sudo

.DELETE_ON_ERROR:
.PHONY: all clean

$(ALPINE_TAR):
	curl -L -o "$@" "https://nl.alpinelinux.org/alpine/v$(ALPINE_MAJOR)/releases/$(ALPINE_ARCH)/alpine-minirootfs-$(ALPINE_VERSION)-$(ALPINE_ARCH).tar.gz"

$(ROOT_FS): $(ALPINE_TAR) $(BUILDENV_SCRIPT)
	dd if=/dev/zero of="$@" count=$(IMAGE_SIZE_MB) bs=1M
	mkfs.ext4 "$@"
	$(ESCALATE_CMD) mkdir -p $(MOUNTPOINT)
	$(ESCALATE_CMD) mount -t ext4 -o loop "$@" $(MOUNTPOINT)
	$(ESCALATE_CMD) tar -C $(MOUNTPOINT) -xvf $(ALPINE_TAR)
	$(ESCALATE_CMD) cp /etc/resolv.conf $(MOUNTPOINT)/etc/resolv.conf
	git submodule init $(SGXLKL_ROOT)/ltp
	git submodule update --remote $(SGXLKL_ROOT)/ltp
	$(ESCALATE_CMD) mkdir $(MOUNTPOINT)/ltp
	$(ESCALATE_CMD) cp -R $(SGXLKL_ROOT)/ltp/* $(MOUNTPOINT)/ltp/
	$(ESCALATE_CMD) install $(BUILDENV_SCRIPT) $(MOUNTPOINT)/usr/sbin
	$(ESCALATE_CMD) chroot $(MOUNTPOINT) /sbin/apk update
	$(ESCALATE_CMD) chroot $(MOUNTPOINT) /sbin/apk add bash


	$(ESCALATE_CMD) chroot $(MOUNTPOINT) /bin/bash /usr/sbin/buildenv.sh 'build' '/ltp/testcases/kernel/syscalls'
	$(ESCALATE_CMD) cp $(MOUNTPOINT)/ltp/.c_binaries_list .
	$(ESCALATE_CMD) umount $(MOUNTPOINT)
	$(ESCALATE_CMD) chown $(USER) "$@"

gettimeout:
	@echo ${LTP_TEST_EXEC_TIMEOUT}

run: run-hw run-sw

run-hw: $(ROOT_FS)
	@${LTP_TEST_SCRIPT} run-hw

run-sw: $(ROOT_FS)
	@${LTP_TEST_SCRIPT} run-sw

run-hw-single: $(ROOT_FS)
	@rm -f $(ROOT_FS_FRESH_COPY)
	@cp $(ROOT_FS) $(ROOT_FS_FRESH_COPY)
	${SGXLKL_STARTER} --hw-debug $(ROOT_FS) $(test)

run-sw-single: $(ROOT_FS)
	@rm -f $(ROOT_FS_FRESH_COPY)
	@cp $(ROOT_FS) $(ROOT_FS_FRESH_COPY)
	${SGXLKL_STARTER} --sw-debug $(ROOT_FS) $(test)

run-hw-single-gdb: $(ROOT_FS)
	@rm -f $(ROOT_FS_FRESH_COPY)
	@cp $(ROOT_FS) $(ROOT_FS_FRESH_COPY)
	${SGXLKL_GDB} --args ${SGXLKL_STARTER} --hw-debug $(ROOT_FS) $(test)

run-sw-single-gdb: $(ROOT_FS)
	@rm -f $(ROOT_FS_FRESH_COPY)
	@cp $(ROOT_FS) $(ROOT_FS_FRESH_COPY)
	${SGXLKL_GDB} --args ${SGXLKL_STARTER} --sw-debug $(ROOT_FS) $(test)

clean:
	@test -f $(ALPINE_TAR) && rm $(ALPINE_TAR) || true
	@test -f $(ROOT_FS) && rm $(ROOT_FS) || true
	@test -f $(ROOT_FS_FRESH_COPY) && rm $(ROOT_FS_FRESH_COPY) || true
	@rm -f .c_binaries_list
```

Reply:
Ah, I see. I was worried about making sure it would continue to work going forward.  I think if we did something like you suggest, it should still have error as a final clause.

I understand now what you mean by "proxy". I didn't really consider it that way.  I considered, can we find these files that happen to be in X location at this time.

I'd like to get other feedback on what others prefer.

Reply:
@SeanTAllen what @letmaik is recommending looks much much cleaner and readable. I absolutely agree with Maik's recommendation.

Reply:
doen